specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+                })
+          );
+
+          return filteredRule;
+        })
+      );
+    },
+    [container.name, loading, tableRules, setTableRules]
+  );
+
+  return (
+    <Dialog
+      buttons={
+        <>
+          <Button.DialogClose>{commonText.close()}</Button.DialogClose>
+          {saveBlocked ? (
+            <Button.Danger
+              className="cursor-not-allowed"
+              onClick={(): void => undefined}
+            >
+              {icons.exclamation}
+              {commonText.save()}
+            </Button.Danger>
+          ) : (
+            <Submit.Save
+              disabled={tableRules === uniquenessRules[model.name]}
+              onClick={(): void => {
+                loading(
+                  ajax(
+                    `/businessrules/uniqueness_rules/${schema.domainLevelIds.discipline}/`,
+                    {
+                      // eslint-disable-next-line @typescript-eslint/naming-convention
+                      headers: { Accept: 'application/json' },
+                      method: tableRules.some(({ rule }) => rule.id === null)
+                        ? 'POST'
+                        : 'PUT',
+                      body: {
+                        rules: tableRules.map(({ rule }) => rule),
+                        model: container.name,
+                      },
+                    }
+                  ).then((): void => {
+                    setStoredTableRules(tableRules);
+                    return isEditing
+                      ? void toggleIsEditing()
+                      : void handleClose();
+                  })
+                );
+              }}
+            >
+              {commonText.save()}
+            </Submit.Save>
+          )}
+        </>
+      }
+      header={header}
+      headerButtons={
+        <>
+          <span className="-ml-2 flex-1" />
+          <Button.Small
+            aria-pressed={isEditing}
+            className="w-fit"
+            disabled={!hasPermission('/schemaconfig/uniquenessrules', 'update')}
+            onClick={toggleIsEditing}
+          >
+            {commonText.edit()}
+          </Button.Small>
+        </>
+      }
+      modal={false}
+      onClose={handleClose}
+    >
+      <table className="grid-table grid-cols-[2fr_1fr] gap-2 gap-y-4 overflow-auto ">
+        <thead>
+          <tr>
+            <td>{schemaText.uniqueFields()}</td>
+            <td>{schemaText.scope()}</td>
+          </tr>
+        </thead>
+        {tableRules?.map(({ rule, duplicates }, index) => (
+          <UniquenessRuleRow
+            fetchedDuplicates={duplicates}
+            fields={fields}
+            isEditing={isEditing}
+            key={index}
+            label={getUniqueInvalidReason(
+              rule.scopes.map(
+                (scope) =>
+                  getFieldsFromPath(model, scope).at(-1) as Relationship
+              ),
+              filterArray(rule.fields.map((field) => model.getField(field)))
+            )}
```

Comment:
bad.
validation message should not be hidden inside of `title`

use useValidation hook instead!

Reply:
Fixed in https://github.com/specify/specify7/pull/4215/commits/66ad3765355550689d875e60f98a1d320d5acc52